shotover-proxy/src/protocols/cassandra_protocol2.rs

```diff
@@ -550,7 +519,7 @@ impl CassandraCodec2 {
             Opcode::Query => {
                 if let Ok(ResponseBody::Query(body)) = frame.get_body() {
                     let parsed_string = CassandraCodec2::parse_query_string(
-                        body.query.clone().into_plain(),
+                        &body.query.clone().into_plain(),
```

Comment:
We can avoid cloning by:
`CassandraCodec2::parse_query_string(body.query.as_str(), &self.pk_col_map);`